src/mio_wrapper.rs

```diff
@@ -0,0 +1,111 @@
+use std::io;
+use std::os::unix::io::{RawFd, FromRawFd};
+
+use libc::{self, c_void};
+use mio::{Poll, Token, Ready, PollOpt, Evented};
+use mio::unix::EventedFd;
+
+pub struct Tun(RawFd);
+pub struct Tap(RawFd);
```

Comment:
It depends on how we will proceed with that.

If we can go with `PollEvented<RawDevice>` then, yes. If we need `Tun(RawDevice)` and `Tap(RawDevice)` -- then it doesn't make sense. We will need to implement Read and Write for Tun, Tap, and RawDevice, i.e. 3 times instead of two times.

It's unclear what will work yet. So I've chosen a simpler approach.